### ISSUE TITLE
Apple Notes Importer doesn't import anything

### ISSUE THREAD

**Issue by axelson** [obsidianmd/obsidian-importer#159]
I installed the Importer plugin today to try to import my apple notes, however nothing is imported and there is a console error (although no user-visible error):

![Screen Shot 2023-10-26 at 8 09 22 PM](https://github.com/obsidianmd/obsidian-importer/assets/9973/50fab14a-21e8-457b-945d-08731e267506)

Here's the full error:
```
Uncaught (in promise) Error: Error: mode should be one of: ascii column csv html insert line list quote tabs tcl

    at Socket.h (plugin:obsidian-importer:71:260)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
h @ plugin:obsidian-importer:71
onceWrapper @ node:events:628
emit @ node:events:513
addChunk @ node:internal/streams/readable:324
readableAddChunk @ node:internal/streams/readable:297
Readable.push @ node:internal/streams/readable:234
onStreamRead @ node:internal/stream_base_commons:190
```

And here's my debug info:
SYSTEM INFO:
	Obsidian version: v1.4.16
	Installer version: v1.4.13
	Operating system: Darwin Kernel Version 20.6.0: Thu Jul  6 22:12:47 PDT 2023; root:xnu-7195.141.49.702.12~1/RELEASE_X86_64 20.6.0
	Login status: logged in
	Catalyst license: supporter
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: none
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 1
	Plugins enabled: 1
		1: Importer v1.5.3

Note: this is on a fresh vault with nothing done except installing the importer plugin.

**Comment by mirnovov** [comment]
It appears that this is since you are on an older unsupported version of macOS which bundles an older version of SQLite, while the [sqlite-tag-spawned](https://github.com/WebReflection/sqlite-tag-spawned) library which the plugin depends on requires a newer version.

Idk whether the Obsidian team supports unsupported versions of macOS, especially as some of the libraries they rely on aren’t supported much further back than this. If they do choose to add compatibility it shouldn’t be too hard to add a fallback for older OS versions though, seems like a relatively easy tweak, and we already bundle this library and modify it.

**Comment by wdfry** [comment]
Same for me on current OS:
 The strange thing, I tried 5 times in my regular vault with plugins enabled, and each import was blank. Tried the sandbox vault and it worked, then went back to mine without closing or reloading or anything and it started working. Below are my logs if it helps:

<img width="549" alt="Screenshot 2023-10-30 at 11 15 56 AM" src="https://github.com/obsidianmd/obsidian-importer/assets/54650211/e18f4bf2-4365-450b-aeaa-6641d0fe76d1">


SYSTEM INFO:
	Obsidian version: v1.4.16
	Installer version: v1.4.16
	Operating system: Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:31 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T8112 23.1.0
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: Minimal v7.3.4
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 15
	Plugins enabled: 15
		1: Advanced Tables v0.19.1
		2: Calendar v1.5.10
		3: Dataview v0.5.63
		4: Editor Syntax Highlight v0.1.3
		5: Find orphaned files and broken links v1.9.1
		6: Homepage v3.6.0
		7: Obsidian42 - BRAT v0.7.0
		8: Quick Switcher++ v3.3.8
		9: QuickAdd v1.4.0
		10: Templater v1.16.4
		11: Datacore v0.1.2
		12: Importer v1.5.3
		13: Minimal Theme Settings v7.3.0
		14: Plugin Update Tracker v1.5.2
		15: Archiver v0.26.1

**Comment by mirnovov** [comment]
This appears to be a separate issue. It seems the `static-params` dependency wasn't upgraded to 0.4 for the build of the latest version, even though the requirements were changed. Hence it fails when `Import recently deleted notes` is checked. I have checked the minified code and the respective changes in 0.4 haven't been incorporated.

**Comment by kepano** [comment]
Fixed in 1.5.4

**Comment by mirnovov** [comment]
@kepano in case you weren't aware, this only fixes wdfry's problem, not the initial problem reported by axelson. Although the symptoms are the same the cause is different (the initial problem being from using an unsupported version of macOS).

**Comment by axelson** [comment]
Ah, issues caused by an older version makes sense (although unfortunately this is the latest OS that this macbook air supports).

Although it would be nice if there was a graphical error message shown, that would help user's know that an error has occurred.

**Comment by mirnovov** [comment]
I've opened a pull request with support for older versions in #163. Though if the Obsidian team chooses not to implement this then a message is a good idea. Keep in mind they are busy (and have already spent quite a bit of time going over my contributions so far) so this may take a while to make it into a release.

**Comment by axelson** [comment]
Awesome, thank you so much!

**Comment by kepano** [comment]
Closed via [1.5.5](https://github.com/obsidianmd/obsidian-importer/releases/tag/1.5.5)
